### PR TITLE
Implement transition animations from region list view to region details view

### PR DESCRIPTION
**Pull request by kellyi**
## Overview

This PR implements the animated transition from the list view to the details view present in the prototype.

To accomplish this I adapted the code in `main.js` to set the animation: essentially it fades the non-selected list items out according to a timer set by index position, then floats the selected list item to the top of the sidebar before hiding it & replacing with a cloned version in the details view.

We replicate the logic here, adding a deferred that we can resolve once the animation's complete to show the details view. Getting this to work required changing how we instantiate the list view a bit: previously we created it once, then showed or hid it programmatically. Now we create it anew each time the list view's shown and destroy it each time the details view is shown. (We were already doing this create/destroy cycle with the details view.)

Connects #6 

## Demo

Prototype animation:

tnc-prototype-animation

This PR animation:

tnc-site-animation

## Testing Instructions
- get this branch, then `server`
- select elements from the list view and verify that you see the animated transition and that the list view's properly re-created on closing the details view
- comment out https://github.com/CoastalResilienceNetwork/tnc-network-site/blob/b3e7f79e332ec6a2468d1d361f6816cf26fb0cbd/src/js/App.js#L78 then reload the app and verify that the list and details views still toggle appropriately but without the animation
- verify that the animation works in IE 11, FF, Chrome, and Safari

### CODE REVIEW

**Comment by kellyi** [comment]
Second commit here removes `main.js` which we kept around for documentary purposes; third commit fixes some improper uses of `_.noop` in the models.js file.

File: src/js/App.js [hunk 1]
```diff
@@ -50,18 +51,61 @@ window.App = Backbone.View.extend({
...
+        this.regionListView = new RegionListView({ model: this.model });
+        this.regionListView.render();
         this.regionListView.$el.fadeIn(200);
     },
 
+    animateDetailsViewReplacingList: function() {
+        var self = this;
+        var selectedRegion = this.model.get('selectedRegion').get('title');
+        var listItems = $('#region-location-list').children();
+        var $sidebarInnerEl = $('.sidebar--inner');
```
Comment (caseycesari) [inline src/js/App.js:63]: If possible, consider replacing these with calls like `this.$el.find()`. It should be marginally faster.

File: src/js/App.js [hunk 2]
```diff
@@ -50,18 +51,61 @@ window.App = Backbone.View.extend({
...
+        promise.then(function() {
+            self.regionListView.$el.hide();
+            self.regionListView.$el.empty();
+            self.regionDetailsView.render();
+            self.regionDetailsView.$el.show();
+        });
+
+        $.each(listItems, function(index, listItem) {
+            if (listItem.id !== selectedRegion) {
+                $(listItem).delay(200 * index).fadeTo(300, 0);
```
Comment (caseycesari) [inline src/js/App.js]: Consider documenting these values.

File: src/js/App.js [hunk 3]
```diff
@@ -50,18 +51,61 @@ window.App = Backbone.View.extend({
...
+            .queue(function(next) {
+                var top = $sidebarInnerEl.offset().top - $selectedListItem.offset().top;
+                $selectedListItem.animate({
+                    top: top + 'px',
+                    margin: '0 -3rem 0 -3rem'
+                }, 500, promise.resolve).fadeOut(500);
+                next();
+            });
+
+        return _.noop();
```
Comment (caseycesari) [inline src/js/App.js]: Consider adding a comment here explaining why we return this.

**Review by caseycesari — APPROVED** [review]
Looks good! Nice job porting this over.

File: src/js/App.js (hunk 3 above)
Comment (kellyi) [reply src/js/App.js]: Sounds good. I'll make a note that this was done to ensure the function has the same return type whether or not the animation code gets executed.

File: src/js/App.js (hunk 1 above)
Comment (kellyi) [reply src/js/App.js:63]: Tried this out and it wasn't able to find the list items to loop through afterward, so it just made the update without the animation. Going to leave as is for now but we can revisit if we encounter performance issues.

**Pull request merged**